src/data/nat/digits.lean

```diff
@@ -236,10 +238,71 @@ begin
   { simp [of_digits, list.sum_cons, ih], }
 end
 
-/-! ### Inequalities
+/-!
+### Properties
+
+This section contains various lemmas of properties relating to `digits` and `of_digits`.
+-/
+
+lemma digits_eq_nil_iff_eq_zero (b n : ℕ) : digits b n = [] ↔ n = 0 :=
+begin
+  split,
+  { intro h,
+    have : of_digits b (digits b n) = of_digits b [], by rw h,
+    convert this,
+    rw of_digits_digits },
+  { rintro rfl,
+    simp }
+end
+
+lemma digits_ne_nil_iff_ne_zero (b n : ℕ) : digits b n ≠ [] ↔ n ≠ 0 :=
+⟨λ h, h ∘ (digits_eq_nil_iff_eq_zero _ _).mpr,
+ λ h, h ∘ (digits_eq_nil_iff_eq_zero _ _).mp⟩
+
+private lemma digits_last_aux1 {α : Type*} {L M : list α} (h : L = M) (p q) : L.last p = M.last q :=
+by simp [h]
 
-This section contains various lemmas with inequalities relating to `digits` and `of_digits`.
- -/
+private lemma digits_last_aux2 (b n : ℕ) (h : 2 ≤ b) (w : 0 < n) :
```

Comment:
```suggestion
private lemma digits_last_aux2 {b n : ℕ} (h : 2 ≤ b) (w : 0 < n) :
```
`b` and `n` can be figured out from later hypotheses, so should be implicit. Please also change this in the other lemmas you added